File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/routes/account/AccountPage.java
package com.chaincuerealestate.chaincuerealestate.routes.account;

import com.chaincuerealestate.chaincuerealestate.domains.Country;
import com.chaincuerealestate.chaincuerealestate.domains.House;
import com.chaincuerealestate.chaincuerealestate.routes.home.HomePage;
import com.chaincuerealestate.chaincuerealestate.routes.home.HomePageDTO;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Optional;
import java.util.function.Function;
import java.util.stream.Stream;

@RestController
@RequestMapping("account")
@RequiredArgsConstructor
@Slf4j
@CrossOrigin(origins = "*")
public class AccountPage {

    @GetMapping
    public ResponseEntity<AccountPageDTO> homePage(@RequestHeader("Authorization") String token) {
        log.info("HomePage");
        log.info(token);
        var toDTO = toHomePageDTO(Optional.empty());
        return ResponseEntity.ok(toDTO);
    }

    private AccountPageDTO toHomePageDTO(Optional<Function<DTOBuilder, DTOBuilder>> additionalProcessing) {
        return Stream.of(new DTOBuilder())
                .map(additionalProcessing.orElseGet(Function::identity))
                .map(AccountPage::toDTO)
                .findFirst().get();
    }

    private static AccountPageDTO toDTO(DTOBuilder dtoBuilder) {
        return new AccountPageDTO("id");
    }

    @Data
    private static class DTOBuilder {

        public DTOBuilder() {
        }
    }

}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/routes/account/AccountPage.kt
package com.example.chaincuerealestate.routes.account

import com.example.chaincuerealestate.routes.home.HomePage
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestHeader
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("account")
class AccountPage {

    @GetMapping()
    fun homePage(@RequestHeader("Authorization") token: String): ResponseEntity<AccountPageDTO> {
        log.info("HomePage")
        log.info(token)
        val toDTO = toHomePageDTO { it }
        return ResponseEntity.ok(toDTO)
    }

    private fun toHomePageDTO(additionalProcessing: ((DTOBuilder) -> DTOBuilder)?): AccountPageDTO {
        return (additionalProcessing?.invoke(DTOBuilder("")) ?: DTOBuilder(""))
            .let { toDTO(it) }
    }

    private fun toDTO(dtoBuilder: DTOBuilder): AccountPageDTO {
        return AccountPageDTO(
            id = dtoBuilder.id
        )
    }

    private data class DTOBuilder(
        val id: String = ""
    )

    private companion object {
        private val log = LoggerFactory.getLogger(HomePage::class.java)
    }
}
